tensorflow/core/platform/hadoop/hadoop_file_system.h

```diff
@@ -78,6 +78,7 @@ class HadoopFileSystem : public FileSystem {
 
  private:
   std::map<std::string, hdfsFS> connectionCache_;
```

Comment:
If you use tensorflow mutex, you can add a macro here 
```cpp
std::map<std::string, hdfsFS> connectionCache_ TF_GUARDED_BY(mu);
```

Reply:
confirm, like this?
hadoop_file_system.h
```cc
 private:
  std::mutex mtx_;
  std::map<std::string, hdfsFS> connectionCache_ TF_GUARDED_BY(mtx_);
  Status Connect(StringPiece fname, hdfsFS* fs);
```
hadoop_file_system.cc
```cc
 Status HadoopFileSystem::Connect(StringPiece fname, hdfsFS* fs) {
  TF_RETURN_IF_ERROR(libhdfs()->status());

  StringPiece scheme, namenode, path;
  io::ParseURI(fname, &scheme, &namenode, &path);
  string nn(namenode);

  string cacheKey(scheme.data(), scheme.size());
  hdfsBuilder* builder = libhdfs()->hdfsNewBuilder();
  if (scheme == "file") {
    libhdfs()->hdfsBuilderSetNameNode(builder, nullptr);
  } else if (scheme == "viewfs") {
    char* defaultFS = nullptr;
    libhdfs()->hdfsConfGetStr("fs.defaultFS", &defaultFS);
    StringPiece defaultScheme, defaultCluster, defaultPath;
    io::ParseURI(defaultFS, &defaultScheme, &defaultCluster, &defaultPath);

    if (scheme != defaultScheme ||
        (namenode != "" && namenode != defaultCluster)) {
      return errors::Unimplemented(
          "viewfs is only supported as a fs.defaultFS.");
    }
    // The default NameNode configuration will be used (from the XML
    // configuration files). See:
    // https://github.com/tensorflow/tensorflow/blob/v1.0.0/third_party/hadoop/hdfs.h#L259
    libhdfs()->hdfsBuilderSetNameNode(builder, "default");
  } else if (scheme == "har") {
    TF_RETURN_IF_ERROR(SplitArchiveNameAndPath(path, nn));
    libhdfs()->hdfsBuilderSetNameNode(builder, nn.c_str());
    cacheKey += nn;
  } else {
    libhdfs()->hdfsBuilderSetNameNode(builder,
                                      nn.empty() ? "default" : nn.c_str());
    cacheKey += nn;
  }
  {
    mutex_lock lock(mtx_);
    if (connectionCache_.find(cacheKey) == connectionCache_.end()) {
      hdfsFS cacheFs = libhdfs()->hdfsBuilderConnect(builder);
      if (cacheFs == nullptr) {
        return errors::NotFound(strerror(errno));
      }
      connectionCache_[cacheKey] = cacheFs;
    }
    *fs = connectionCache_[cacheKey];
  }
  return Status::OK();
}
```

Reply:
@vnvo2409
Can you help me review the PR again? Thanks!

Reply:
The idiomatic name should be `mu_`.

Reply:
LGTM. Thanks for the prompt fix!